app/src/app/data_model.cljc

```diff
@@ -34,7 +34,7 @@
 (s/defschema NonEmptyString (s/constrained s/Str seq "Non-empty string"))
 
 (s/defschema EmailAddress (s/constrained NonEmptyString
-                                         (fn [s] (re-matches #"[^@]+@[^@]+" s))
+                                         (fn [s] (re-matches #"^\s*[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}\s*$" s))
```

Comment:
are you sure this covers everything? :)

> [...] In addition to the above ASCII characters, international characters above U+007F, encoded as UTF-8, are permitted by RFC 6531, [...]

Seems to me that this regex doesn't support Umlaute, which are allowed since 2012.